File: 0841-keys-and-rooms/0841-keys-and-rooms.java
class Solution {
    public boolean canVisitAllRooms(List<List<Integer>> rooms) {
        boolean[] isVisited = new boolean[rooms.size()];
        // 첫 번째 방부터 탐색
        dfs(rooms, isVisited, 0);
            
        for (int i = 0; i < isVisited.length; i++) {
            if (isVisited[i] == false) return false;
        }

        return true;
    }

    public void dfs(List<List<Integer>> rooms, boolean[] isVisited, int idx) {
        if (isVisited[idx]) return;
        else isVisited[idx] = true;

        for (int key : rooms.get(idx)) {
            if (isVisited[key] == false) {
                dfs(rooms, isVisited, key);
            }
        }    
    }
} 

File: 0841-keys-and-rooms/0841-keys-and-rooms.py
class Solution:
    def canVisitAllRooms(self, rooms: List[List[int]]) -> bool:
        output = 0
        visited = [False] * len(rooms)
        self.dfs(rooms, visited, 0)
        
        for i in range(len(visited)):
            if visited[i] is False:
                return False
        
        return True

    def dfs(self, rooms, visited, key):
        print(key)
        if visited[key] is True:
            return
        visited[key] = True

        for i in range(len(rooms[key])):
            if visited[rooms[key][i]] is False:
                self.dfs(rooms, visited, rooms[key][i]) 
    
        